src/shared/components/home/emojis-form.tsx

```diff
@@ -0,0 +1,456 @@
+import { Component } from "inferno";
+import { GetSiteResponse, UserOperation, wsJsonToRes, wsUserOp } from "lemmy-js-client";
+import { CreateCustomEmoji, CustomEmojiResponse, DeleteCustomEmoji, DeleteCustomEmojiResponse, EditCustomEmoji } from "lemmy-js-client/dist/interfaces/api/custom_emoji";
+import { WebSocketService } from "../../services";
+import { i18n } from "../../i18next";
+import { customEmojisLookup, isBrowser, myAuth, removeFromEmojiDataModel, setIsoData, toast, updateEmojiDataModel, wsClient, wsSubscribe } from "../../utils";
+import { EmojiMart } from "../common/emoji-mart";
+import { HtmlTags } from "../common/html-tags";
+import { Icon } from "../common/icon";
+import { Subscription } from "rxjs";
+import { pictrsUri } from "../../env";
+import { Paginator } from "../common/paginator";
+
+interface EmojiFormState {
+    siteRes: GetSiteResponse;
+    customEmojis: CustomEmojiViewForm[];
+    loading: boolean;
+    page: number;
+}
+
+class CustomEmojiViewForm {
+    id: number;
+    category: string;
+    shortcode: string;
+    image_url: string;
+    alt_text: string;
+    keywords: string;
+    changed: boolean;
+    page: number;
+}
+
+export class EmojiForm extends Component<any, EmojiFormState> {
+    private isoData = setIsoData(this.context);
+    private subscription: Subscription | undefined;
+    private itemsPerPage = 15;
+    private emptyState: EmojiFormState = {
+        loading: false,
+        siteRes: this.isoData.site_res,
+        customEmojis: this.isoData.site_res.custom_emojis.map((x, index) =>
+        ({
+            id: x.custom_emoji.id,
+            category: x.custom_emoji.category,
+            shortcode: x.custom_emoji.shortcode,
+            image_url: x.custom_emoji.image_url,
+            alt_text: x.custom_emoji.alt_text,
+            keywords: x.keywords.map(x => x.keyword).join(" "),
+            changed: false,
+            page: 1 + (Math.floor(index / this.itemsPerPage))
```

Comment:
Don't hack in paging to the front end, if it needs to be added, it needs to be done properly in the back end as well.